### PR TITLE
fix issue 397: move check_vm_settings.yml to be executed after revert…

### PR DESCRIPTION
**Pull request by ZouYuhua**
change code as https://github.com/vmware/ansible-vsphere-gos-validation/issues/397 required.

### CODE REVIEW

File: env_setup/env_setup.yml [hunk 1]
```diff
@@ -34,6 +34,9 @@
         fail_msg: "VM '{{ vm_name }}' doesn't exist. Please set new_vm to True to deploy the VM or provide an existing VM name."
       when: new_vm is undefined or not new_vm | bool
 
+    # Check VM settings
+    - include_tasks: check_vm_settings.yml
```
Comment (keirazhang) [inline env_setup/env_setup.yml:38]: Here needs a task name.

File: env_setup/env_setup.yml [hunk 2]
```diff
@@ -92,8 +95,13 @@
...
-        # Check VM settings
-        - include_tasks: check_vm_settings.yml
+        - name: "Get VM power state"
+          include_tasks: ../../common/vm_get_power_state.yml
+
+        - name: "Poweron VM"
+          include_tasks: ../../common/vm_set_power_state.yml
+          vars:
+            vm_power_state_set: "powered-on"
+          when: vm_power_state_get == "poweredOff"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:106]: cleanup snapshot and power on VM shall happens before check_vm_settings.yml

File: env_setup/env_setup.yml [hunk 3]
```diff
@@ -6,20 +6,24 @@
...
   gather_facts: false
   tasks:
-    - include_tasks: get_ansible_info.yml
-    - include_tasks: set_default_variables.yml
+    - name: "Get ansible info"
+      include_tasks: get_ansible_info.yml
 
-    # Create this test run log files path
-    - include_tasks: create_local_log_path.yml
+    - name: "Get default variables"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:12]: Get -> Set

File: env_setup/env_setup.yml [hunk 4]
```diff
@@ -34,27 +38,20 @@
...
-      when: vm_exists is defined and vm_exists
-    - name: Set fact of the VM datastore path
-      ansible.builtin.set_fact:
-        vm_datastore_path: "/vmfs/volumes/{{ datastore }}"
-    - name: Display the datatore path of VM files
-      ansible.builtin.debug: var=vm_datastore_path
-
-    # Get vCenter and ESXi version info
-    - include_tasks: ../common/vcenter_get_version_build.yml
+    - name: "Get vCenter and ESXi version info"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:41]: Change name to "Get vCenter Server version and build"

File: env_setup/env_setup.yml [hunk 5]
```diff
@@ -67,8 +64,41 @@
...
-    # Add esxi host into memory inventory
-    - include_tasks: ../common/add_host_in_memory_inventory.yml
+    - name: "Cleanup snapshots and get VM related information"
+      block:
+        - name: "Not cleanup old snapshot by default"
+          ansible.builtin.set_fact:
+            cleanup_old_snapshots: false
+          when: cleanup_old_snapshots is undefined
+
+        - name: "Cleanup existing snapshots of existing VM if required"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:74]: Change name to "Revert and cleanup existing snapshots on existing VM if required"

File: env_setup/env_setup.yml [hunk 6]
```diff
@@ -67,8 +64,41 @@
...
+          include_tasks: ../common/vm_cleanup_snapshot.yml
+
+        - name: "Get VM power state"
+          include_tasks: ../../common/vm_get_power_state.yml
+
+        - name: "Poweron VM"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:80]: Poweron -> Power on

File: env_setup/env_setup.yml [hunk 7]
```diff
@@ -67,8 +64,41 @@
...
+          include_tasks: ../../common/vm_set_power_state.yml
+          vars:
+            vm_power_state_set: "powered-on"
+          when: vm_power_state_get == "poweredOff"
+
+        - name: "Check VM settings"
+          include_tasks: check_vm_settings.yml
+
+        - name: "Get VM related info if testing on existing VM"
```
Comment (keirazhang) [inline env_setup/env_setup.yml:89]: Change name to "Get existing VM info"

File: env_setup/env_setup.yml [hunk 8]
```diff
@@ -67,8 +64,41 @@
...
+        - name: "Not cleanup old snapshot by default"
+          ansible.builtin.set_fact:
+            cleanup_old_snapshots: false
+          when: cleanup_old_snapshots is undefined
+
+        - name: "Cleanup existing snapshots of existing VM if required"
+          include_tasks: ../common/vm_cleanup_snapshot.yml
+
+        - name: "Get VM power state"
+          include_tasks: ../../common/vm_get_power_state.yml
```
Comment (keirazhang) [inline env_setup/env_setup.yml:78]: ../../common -> ../common

File: env_setup/env_setup.yml [hunk 9]
```diff
@@ -67,8 +64,41 @@
...
+
+        - name: "Poweron VM"
+          include_tasks: ../../common/vm_set_power_state.yml
```
Comment (keirazhang) [inline env_setup/env_setup.yml:81]: ../../common -> ../common

File: env_setup/env_setup.yml (hunk 3 above)
Comment (ZouYuhua) [reply env_setup/env_setup.yml]: Thanks. change it.

File: env_setup/env_setup.yml (hunk 6 above)
Comment (ZouYuhua) [reply env_setup/env_setup.yml]: Thanks. change it.

File: env_setup/env_setup.yml (hunk 8 above)
Comment (ZouYuhua) [reply env_setup/env_setup.yml]: Thanks. change it.

File: env_setup/env_setup.yml (hunk 7 above)
Comment (ZouYuhua) [reply env_setup/env_setup.yml]: Thanks. change it.

File: env_setup/env_setup.yml (hunk 5 above)
Comment (ZouYuhua) [reply env_setup/env_setup.yml]: Thanks. change it.

**Comment by ZouYuhua** [comment]
Pass for the testing after codereview with pardus 21.5 xfce:
image

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**